Refs CVE-2022-34265 -- Properly escaped Extract() and Trunc() parameters. (#15820)
django/django: docs/releases/4.1.txt

  ``DatabaseOperations.insert_statement()`` method is replaced by
  ``on_conflict`` that accepts ``django.db.models.constants.OnConflict``.

> * The ``DatabaseOperations.date_extract_sql()``,
>   ``DatabaseOperations.datetime_extract_sql()``,
>   ``DatabaseOperations.time_extract_sql()``,
>   ``DatabaseOperations.date_trunc_sql()``,
>   ``DatabaseOperations.datetime_trunc_sql()``,
>   ``DatabaseOperations.time_trunc_sql()``,
>   ``DatabaseOperations.datetime_cast_date_sql()``, and
>   ``DatabaseOperations.datetime_cast_time_sql()`` methods now take ``sql`` and
>   ``params`` arguments instead of ``field_name`` and return 2-tuple containing
>   some SQL and the parameters to be interpolated into that SQL. The new
>   signatures are as follows:

Reviewer: Awkward to open the sentence with such a long list, makes it hard to spot what has actually changed. I suggest only listing the names below:

```suggestion
* Several date and time methods on ``DatabaseOperations`` now take ``sql`` and
  ``params`` arguments instead of ``field_name``, and return a 2-tuple of
  SQL and the parameters to be interpolated into that SQL. The changed
  methods have these new signatures:
```
